sql/plan.go

```diff
@@ -50,6 +49,10 @@ func (p *planner) resetTxn() {
 // plan needs to be iterated over using planNode.Next() and planNode.Values()
 // in order to retrieve matching rows.
 func (p *planner) makePlan(stmt parser.Statement) (planNode, error) {
+	if stmt.StatementType() == parser.DDL {
+		p.txn.SetSystemDBTrigger()
```

Comment:
I think that's OK; those statements will bail out early and won't actually do anything. The only exception is `BEGIN; CREATE TABLE IF NOT EXISTS; COMMIT;` which will exhibit the behaviour you describe. I think it's OK and I'll add a comment. cc @mberhault